hardware_interface/test/interface_manager_test.cpp

```diff
@@ -84,6 +84,93 @@ TEST(InterfaceManagerTest, InterfaceRewriting)
   EXPECT_EQ(2, foo_iface_ptr->foo);
 }
 
+struct TestCombinationHandle {
+  TestCombinationHandle(const std::string& name) : name{name} {}
+
+  std::string getName() const
+  {
+    return name;
+  }
+
+  std::string name;
+};
+
+struct TestCombinationInterface : public ResourceManager<TestCombinationHandle>
+{
+
+};
+
+TEST(InterfaceManagerTest, InterfaceCombination)
+{
+  // Create two InterfaceManagers, each containing a simple interface and a ResourceManager interface,
+  // and an additional simple interface for the second one
+  InterfaceManager root_mgr;
+  InterfaceManager leaf_mgr;
+
+  FooInterface foo_root_iface(1);
+  root_mgr.registerInterface(&foo_root_iface);
+  FooInterface foo_leaf_iface(2);
+  leaf_mgr.registerInterface(&foo_leaf_iface);
+
+  BazInterface baz_leaf_iface;
+  baz_leaf_iface.baz = 4.2;
+  leaf_mgr.registerInterface(&baz_leaf_iface);
+
+  TestCombinationInterface combi_root_iface;
+  TestCombinationHandle combi_root_handle("foo_handle");
+  combi_root_iface.registerHandle(combi_root_handle);
+  root_mgr.registerInterface(&combi_root_iface);
+
+  TestCombinationInterface combi_leaf_iface;
+  TestCombinationHandle combi_leaf_handle("bar_handle");
+  combi_leaf_iface.registerHandle(combi_leaf_handle);
+  leaf_mgr.registerInterface(&combi_leaf_iface);
```

Comment:
Can you rename these handles from `"foo_handle"` and `"bar_handle"`? I think it causes confusion with the `FooInterface` above. Maybe just match the var name `"combi_root_handle"` and `"combi_leaf_handle"`?

Reply:
Agreed, that makes much more sense. Applied in rebased commits.